pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -88,11 +88,18 @@ func LoadBalancerSrcRanges() []string {
 
 // GetLoadBalancer is an implementation of LoadBalancer.GetLoadBalancer
 func (gce *GCECloud) GetLoadBalancer(clusterName string, service *v1.Service) (*v1.LoadBalancerStatus, bool, error) {
-	loadBalancerName := cloudprovider.GetLoadBalancerName(service)
+	// For backword compatibility we had to fetch the loadBalancerName this way or else a simple assignment for service.Status.LoadBalancer.Name would be enough.
+	var loadBalancerName string
+	if len(service.Status.LoadBalancer.Name) > 0 {
+		loadBalancerName = service.Status.LoadBalancer.Name
```

Comment:
Excellent point.  This logic (everywhere) needs to look more like:

```
if the status name exists, use that
else check if the legacy name exists
else check if the ideal name exists
```

and when writing back, always write the status name